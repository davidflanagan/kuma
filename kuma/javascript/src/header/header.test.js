//@flow
import React from 'react';
import { create } from 'react-test-renderer';
import DocumentProvider from '../document-provider.jsx';
import { fakeDocumentData } from '../document-provider.test.js';
import Header from './header.jsx';

test('Header snapshot', () => {
    const header = create(
        <DocumentProvider initialDocumentData={fakeDocumentData}>
            <Header />
        </DocumentProvider>
    );
    expect(header.toJSON()).toMatchSnapshot();
});
