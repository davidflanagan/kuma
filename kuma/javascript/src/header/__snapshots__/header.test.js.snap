// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
.emotion-46 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-top: 4px solid #83d0f2;
  height: 64px;
  box-sizing: border-box;
}

.emotion-0 {
  display: block;
  height: 60px;
}

.emotion-1 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 4px;
  -ms-flex-preferred-size: 4px;
  flex-basis: 4px;
}

.emotion-9 {
  position: relative;
  pointer-events: auto;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  font-weight: bold;
  line-height: 32px;
  white-space: nowrap;
  padding: 0 10px;
  margin: 0;
  border: none;
}

.emotion-5:hover {
  background-color: #eee;
}

.emotion-5:focus {
  outline-offset: -3px;
}

.emotion-3 {
  font-size: 12px;
  padding-left: 2px;
}

.emotion-7 {
  position: absolute;
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  background-color: white;
  border: solid #83d0f2 1.5px;
  box-shadow: 3px 3px 5px rgba(0,0,0,0.25);
  padding: 10px;
  min-width: 100%;
  display: none;
}

.emotion-7 li {
  padding: 5px;
  white-space: nowrap;
}

.emotion-35 {
  -webkit-flex: 1 1 0px;
  -ms-flex: 1 1 0px;
  flex: 1 1 0px;
}

.emotion-39 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2px 8px;
  border: solid #888 1px;
  border-radius: 8px;
  max-width: 350px;
  min-width: 120px;
  -webkit-flex: 2 1 120px;
  -ms-flex: 2 1 120px;
  flex: 2 1 120px;
}

.emotion-37 {
  fill: #333;
  -webkit-flex: 0 0 20px;
  -ms-flex: 0 0 20px;
  flex: 0 0 20px;
}

.emotion-38 {
  border-width: 0 !important;
  font-size: 16px;
  font-weight: bold !important;
  -webkit-flex: 0 1 300px;
  -ms-flex: 0 1 300px;
  flex: 0 1 300px;
}

.emotion-40 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 15px;
  -ms-flex-preferred-size: 15px;
  flex-basis: 15px;
}

.emotion-43 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 18px;
  font-weight: bold;
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  line-height: 48px;
  padding: 0 8px;
}

.emotion-43:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #eee;
}

.emotion-43:focus {
  outline-offset: -3px;
}

.emotion-42 {
  margin-left: 5px;
}

<div
  className="emotion-46 emotion-47"
>
  <a
    className="emotion-0"
    href="/en-US/"
  >
    <svg mock
      alt="MDN Web Docs Logo"
    />
  </a>
  <div
    className="emotion-1 emotion-2"
    width={4}
  />
  <div
    className="emotion-9 emotion-10"
  >
    <button
      className="emotion-5 emotion-6"
      onClick={[Function]}
    >
      Technologies
      <span
        className="emotion-3 emotion-4"
      >
        ‚ñº
      </span>
    </button>
    <ul
      className="dropdown-menu emotion-7 emotion-8"
    >
      <li>
        <a
          href="/en-US/docs/Web/HTML"
        >
          HTML
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/Web/CSS"
        >
          CSS
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/Web/JavaScript"
        >
          JavaScript
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/Web/Guide/Graphics"
        >
          Graphics
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/Web/HTTP"
        >
          HTTP
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/Web/API"
        >
          APIs / DOM
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/Mozilla/Add-ons/WebExtensions"
        >
          Browser Extensions
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/Web/MathML"
        >
          MathML
        </a>
      </li>
    </ul>
  </div>
  <div
    className="emotion-9 emotion-10"
  >
    <button
      className="emotion-5 emotion-6"
      onClick={[Function]}
    >
      References & Guides
      <span
        className="emotion-3 emotion-4"
      >
        ‚ñº
      </span>
    </button>
    <ul
      className="dropdown-menu emotion-7 emotion-8"
    >
      <li>
        <a
          href="/en-US/docs/Learn"
        >
          Learn web development
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/Web/Tutorials"
        >
          Tutorials
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/Web/Reference"
        >
          References
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/Web/Guide"
        >
          Developer Guides
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/Web/Accessibility"
        >
          Accessibility
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/Games"
        >
          Game development
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/Web"
        >
          ...more docs
        </a>
      </li>
    </ul>
  </div>
  <div
    className="emotion-9 emotion-10"
  >
    <button
      className="emotion-5 emotion-6"
      onClick={[Function]}
    >
      Feedback
      <span
        className="emotion-3 emotion-4"
      >
        ‚ñº
      </span>
    </button>
    <ul
      className="dropdown-menu emotion-7 emotion-8"
    >
      <li>
        <a
          href="https://support.mozilla.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Get Firefox help
           üåê
        </a>
      </li>
      <li>
        <a
          href="https://stackoverflow.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Get web development help
           üåê
        </a>
      </li>
      <li>
        <a
          href="/en-US/docs/MDN/Community"
        >
          Join the MDN community
        </a>
      </li>
      <li>
        <a
          href="https://github.com/mdn/sprints/issues/new?template=issue-template.md&projects=mdn/sprints/2&labels=user-report&title=http%3A%2F%2Flocalhost%2F"
          rel="noopener noreferrer"
          target="_blank"
        >
          Report a content problem
           üåê
        </a>
      </li>
      <li>
        <a
          href="https://bugzilla.mozilla.org/form.mdn"
          rel="noopener noreferrer"
          target="_blank"
        >
          Report a bug
           üåê
        </a>
      </li>
    </ul>
  </div>
  <div
    className="emotion-9 emotion-10"
  >
    <button
      className="emotion-5 emotion-6"
      onClick={[Function]}
    >
      <svg mock />
      <span
        className="emotion-3 emotion-4"
      >
        ‚ñº
      </span>
    </button>
    <ul
      className="dropdown-menu emotion-7 emotion-8"
    >
      <li
        lang="es"
      >
        <a
          href="[fake spanish url]"
          title="Spanish"
        >
          <bdi>
            Espa√±ol
          </bdi>
        </a>
      </li>
    </ul>
  </div>
  <div
    className="emotion-35 emotion-36"
  />
  <form
    action="/en-US/search"
    className="emotion-39"
    id="nav-main-search"
    method="get"
    role="search"
  >
    <svg mock
      className="emotion-37"
    />
    <input
      className="emotion-38"
      id="main-q"
      name="q"
      placeholder="Search MDN"
      type="search"
    />
  </form>
  <div
    className="emotion-40 emotion-2"
    width={15}
  />
  <a
    className="emotion-43"
    data-service="GitHub"
    href="/users/github/login/?next=/"
    rel="nofollow"
  >
    Sign in
     
    <svg mock
      className="emotion-42"
    />
  </a>
  <div
    className="emotion-40 emotion-2"
    width={15}
  />
</div>
`;
